File: jdi-uitest-tutorialtests/src/test/java/com/epam/jdi/uitests/testing/career/common_tests/TestsBase.java
package com.epam.jdi.uitests.testing.career.common_tests;

import com.epam.jdi.uitests.testing.career.page_objects.site.EpamSite;
import com.epam.jdi.uitests.web.testng.testRunner.TestNGBase;
import org.testng.annotations.BeforeSuite;

import static com.epam.jdi.uitests.core.settings.JDISettings.*;
import static com.epam.jdi.uitests.testing.career.page_objects.site.EpamSite.homePage;
import static com.epam.jdi.uitests.web.selenium.elements.composite.Site.Init;

/**
 * Created by Roman_Iovlev on 7/13/2015.
 */
public abstract class TestsBase extends TestNGBase {

    @BeforeSuite(alwaysRun = true)
    public static void setUp() throws Exception {
        driverFactory.setDriverPath("C:\\Selenium");
        initJDIFromProperties();
        Init(EpamSite.class);
        homePage.open();
        logger.info("Run Tests");
    }
}

File: C#.Net/Tests/JDI Tests/Scenarios/Tests/TestsBase.cs
using JDI_Commons;
using JDI_Tests.Scenarios.Page_Objects;
using JDI_Web.Selenium.DriverFactory;
using JDI_Web.Selenium.Elements.Composite;
using NUnit.Framework;
using static Epam.JDI.Core.Settings.JDISettings;
using static JDI_Web.Selenium.DriverFactory.WinProcUtils;
using static JDI_Web.Settings.WebSettings;

namespace JDI_Tests.Scenarios.Tests
{
    [SetUpFixture]
    public class TestsBase
    {
        Timer _timer = new Timer();

        [OneTimeSetUp]
        public void Init()
        {
            InitFromProperties();
            Logger.Info("Init test run");
            KillAllRunWebDrivers();

            if (!DriverFactory.HasDrivers())
                UseDriver(DriverTypes.Chrome);
            _timer = new Timer();
            
            WebSite.Init(typeof(EpamSite));
            EpamSite.HomePage.Open();
            Logger.Info("Run Tests");
        }

        [OneTimeTearDown]
        public void TestCleanup()
        {
            Logger.Info($@"
Test run finished.
Total test run time: {_timer.TimePassed.ToString(@"hh\:mm\:ss\.fff")}");
            KillAllRunWebDrivers();
        }
    }
    
}
